complexify-android/src/main/java/co/infinum/complexify/Complexify.kt

```diff
@@ -0,0 +1,95 @@
+package co.infinum.complexify
+
+import co.infinum.complexify.Constants.CHARSETS_ARRAY
+import co.infinum.complexify.Constants.DEFAULT_BAN_LIST
+import kotlin.math.ln
+import kotlin.math.pow
+
+/**
+ * Class for calculation of password complexity.
+ *
+ * @param banMode             Use strict or loose comparisons for banned passwords.
+ * @param strengthScaleFactor Required password strength multiplier. Bigger factor = more complex password required.
+ * @param minimumChars        Minimum password length.
+ * @param banList             A list of banned passwords which will reduce input password's complexity to 0 if the password contains one of them. The ban list is case-insensitive. The default values are generated from 500 worst passwords and 370 Banned Twitter lists: https://wiki.skullsecurity.org/Passwords
+ */
+class Complexify @JvmOverloads constructor(
+    var banMode: ComplexifyBanMode = ComplexifyBanMode.STRICT,
+    var strengthScaleFactor: Double = 1.0,
+    var minimumChars: Int = 8,
+    var banList: Array<String> = DEFAULT_BAN_LIST
```

Comment:
Rule is, when there are more than 3 input params - create a DTO class or interface for it.
Monads are way better than triads, everything more is bad.